In [1]:
# import libraries
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.feature_selection import SelectKBest, mutual_info_regression
from sklearn.ensemble import RandomForestRegressor
import matplotlib.pyplot as plt
import os

In [2]:
os.getcwd()

'/Users/sromee/Documents/AIPI-520/ai-520-wthr-proj'

In [3]:
df_2024 = pd.read_csv("wthr_dt_2024.csv", low_memory = False)
df_2024.head()

,STATION,DATE,LATITUDE,LONGITUDE,ELEVATION,NAME,REPORT_TYPE,SOURCE,HourlyAltimeterSetting,HourlyDewPointTemperature,...,BackupDirection,BackupDistance,BackupDistanceUnit,BackupElements,BackupElevation,BackupEquipment,BackupLatitude,BackupLongitude,BackupName,WindEquipmentChangeDate
0,72306013722,2024-01-01T00:51:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-15,7,30.03,32,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2009-07-08
1,72306013722,2024-01-01T01:00:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-12,4,NaN,32,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2009-07-08
2,72306013722,2024-01-01T01:51:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-15,7,30.02,31,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2009-07-08
3,72306013722,2024-01-01T02:51:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-15,7,30.03,30,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2009-07-08
4,72306013722,2024-01-01T03:51:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-15,7,30.01,30,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,2009-07-08


In [4]:
df_2025 = pd.read_csv("wthr_dt_2025.csv", low_memory = False)
df_2025.head()

,STATION,DATE,LATITUDE,LONGITUDE,ELEVATION,NAME,REPORT_TYPE,SOURCE,HourlyAltimeterSetting,HourlyDewPointTemperature,...,BackupDirection,BackupDistance,BackupDistanceUnit,BackupElements,BackupElevation,BackupEquipment,BackupLatitude,BackupLongitude,BackupName,WindEquipmentChangeDate
0,USW00013722,2025-01-01T00:00:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",SOD,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,07/08/2009
1,USW00013722,2025-01-01T00:51:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-15,343.0,1005.4,5.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,07/08/2009
2,USW00013722,2025-01-01T01:00:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-12,223.0,NaN,5.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,07/08/2009
3,USW00013722,2025-01-01T01:51:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-15,343.0,1006.1,3.3,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,07/08/2009
4,USW00013722,2025-01-01T02:51:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-15,343.0,1007.5,2.2,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,07/08/2009


In [5]:
# checking that both years' columns have the same number of columns
df_2024.shape, df_2025.shape

((13214, 125), (9831, 125))

In [6]:
# combining data from both 2024 & 2025 into 1 df

combined_df = pd.concat([df_2024, df_2025], axis=0, ignore_index=True)
df = combined_df
df.tail()

,STATION,DATE,LATITUDE,LONGITUDE,ELEVATION,NAME,REPORT_TYPE,SOURCE,HourlyAltimeterSetting,HourlyDewPointTemperature,...,BackupName,WindEquipmentChangeDate,MonthlyAverageWindSpeed,CoolingDegreeDaysSeasonToDate,MonthlyCoolingDegreeDays,MonthlyNumberDaysWithSnowfall,HeatingDegreeDaysSeasonToDate,MonthlyHeatingDegreeDays,MonthlyNumberDaysWithThunderstorms,MonthlyNumberDaysWithHeavyFog
23040,USW00013722,2025-09-29T13:00:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-12,223.0,NaN,19.4,...,NaN,07/08/2009,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
23041,USW00013722,2025-09-29T13:51:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-15,223.0,1020.0,19.4,...,NaN,07/08/2009,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
23042,USW00013722,2025-09-29T14:51:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-15,223.0,1019.6,20.0,...,NaN,07/08/2009,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
23043,USW00013722,2025-09-29T15:51:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-15,223.0,1019.6,20.0,...,NaN,07/08/2009,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
23044,USW00013722,2025-09-29T16:00:00,35.8923,-78.7819,126.8,"RALEIGH AIRPORT, NC US",FM-12,223.0,NaN,20.0,...,NaN,07/08/2009,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


# Deciding the response variable:

Since the project goal is to predict the actual air temperature, the column 'HourlyDryBulbTemperature' is the response variable.

The column 'HourlyWetBulbTemperature' is a derived measure made up of combining temperature and humidity and is typically better suited as a feature, not as a target.

# Removing columns with many missing values or that are redundant:

There are 125 columns, which is a large number. So, many will have to be removed to simplify the scope of the project. One criteria is the number of missing values. If columns have > 50% of missing values, they will be removed, especially if their same information is found in other columns.

In [7]:
threshold = 0.5
df = df.loc[:, df.isnull().mean() < threshold]


In [8]:
print("Remaining columns:", len(df.columns))
print(df.columns.tolist())


Remaining columns: 22
['STATION', 'DATE', 'LATITUDE', 'LONGITUDE', 'ELEVATION', 'NAME', 'REPORT_TYPE', 'SOURCE', 'HourlyAltimeterSetting', 'HourlyDewPointTemperature', 'HourlyDryBulbTemperature', 'HourlyPrecipitation', 'HourlyRelativeHumidity', 'HourlySkyConditions', 'HourlySeaLevelPressure', 'HourlyStationPressure', 'HourlyVisibility', 'HourlyWetBulbTemperature', 'HourlyWindDirection', 'HourlyWindSpeed', 'REM', 'WindEquipmentChangeDate']


In [9]:
df.head()

,STATION,DATE,LATITUDE,LONGITUDE,ELEVATION,NAME,REPORT_TYPE,SOURCE,HourlyAltimeterSetting,HourlyDewPointTemperature,...,HourlyRelativeHumidity,HourlySkyConditions,HourlySeaLevelPressure,HourlyStationPressure,HourlyVisibility,HourlyWetBulbTemperature,HourlyWindDirection,HourlyWindSpeed,REM,WindEquipmentChangeDate
0,72306013722,2024-01-01T00:51:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-15,7,30.03,32,...,68.0,SCT:04 210,30.03,29.56,10.00,38.0,230,5.0,MET11701/01/24 00:51:02 METAR KRDU 010551Z 230...,2009-07-08
1,72306013722,2024-01-01T01:00:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-12,4,NaN,32,...,68.0,NaN,30.03,29.57,9.94,38.0,230,5.0,SYN08072306 32966 42304 10056 20000 30013 4016...,2009-07-08
2,72306013722,2024-01-01T01:51:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-15,7,30.02,31,...,67.0,BKN:07 230,30.02,29.55,10.00,37.0,230,3.0,MET10001/01/24 01:51:02 METAR KRDU 010651Z 230...,2009-07-08
3,72306013722,2024-01-01T02:51:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-15,7,30.03,30,...,68.0,BKN:07 220,30.02,29.56,10.00,36.0,240,5.0,MET10001/01/24 02:51:02 METAR KRDU 010751Z 240...,2009-07-08
4,72306013722,2024-01-01T03:51:00,35.89227,-78.78194,120.5,"RALEIGH DURHAM INTERNATIONAL AIRPORT, NC US",FM-15,7,30.01,30,...,73.0,BKN:07 220,30.01,29.54,10.00,35.0,240,5.0,MET10601/01/24 03:51:02 METAR KRDU 010851Z 240...,2009-07-08


In [10]:
df.columns

Index(['STATION', 'DATE', 'LATITUDE', 'LONGITUDE', 'ELEVATION', 'NAME',
       'REPORT_TYPE', 'SOURCE', 'HourlyAltimeterSetting',
       'HourlyDewPointTemperature', 'HourlyDryBulbTemperature',
       'HourlyPrecipitation', 'HourlyRelativeHumidity', 'HourlySkyConditions',
       'HourlySeaLevelPressure', 'HourlyStationPressure', 'HourlyVisibility',
       'HourlyWetBulbTemperature', 'HourlyWindDirection', 'HourlyWindSpeed',
       'REM', 'WindEquipmentChangeDate'],
      dtype='object')

Some columns have the same value for every entry, so they are redundant & can be safely removed

In [11]:
df = df.drop(['STATION', 'LATITUDE', 'LONGITUDE', 'ELEVATION', 'NAME', 'WindEquipmentChangeDate'], axis=1)
df.columns

Index(['DATE', 'REPORT_TYPE', 'SOURCE', 'HourlyAltimeterSetting',
       'HourlyDewPointTemperature', 'HourlyDryBulbTemperature',
       'HourlyPrecipitation', 'HourlyRelativeHumidity', 'HourlySkyConditions',
       'HourlySeaLevelPressure', 'HourlyStationPressure', 'HourlyVisibility',
       'HourlyWetBulbTemperature', 'HourlyWindDirection', 'HourlyWindSpeed',
       'REM'],
      dtype='object')

Other variables can be removed because their values are ambiguous or hard to interpret. For instance, some values in column 'HourlySkyConditions' are similar to:

FEW:02 36.58

The column 'REM' has similarly ambiguous values.


In [12]:
df = df.drop(['HourlySkyConditions', 'REM'], axis=1)
df.columns

Index(['DATE', 'REPORT_TYPE', 'SOURCE', 'HourlyAltimeterSetting',
       'HourlyDewPointTemperature', 'HourlyDryBulbTemperature',
       'HourlyPrecipitation', 'HourlyRelativeHumidity',
       'HourlySeaLevelPressure', 'HourlyStationPressure', 'HourlyVisibility',
       'HourlyWetBulbTemperature', 'HourlyWindDirection', 'HourlyWindSpeed'],
      dtype='object')

In [13]:
df.head()

,DATE,REPORT_TYPE,SOURCE,HourlyAltimeterSetting,HourlyDewPointTemperature,HourlyDryBulbTemperature,HourlyPrecipitation,HourlyRelativeHumidity,HourlySeaLevelPressure,HourlyStationPressure,HourlyVisibility,HourlyWetBulbTemperature,HourlyWindDirection,HourlyWindSpeed
0,2024-01-01T00:51:00,FM-15,7,30.03,32,42,0.00,68.0,30.03,29.56,10.00,38.0,230,5.0
1,2024-01-01T01:00:00,FM-12,4,NaN,32,42,NaN,68.0,30.03,29.57,9.94,38.0,230,5.0
2,2024-01-01T01:51:00,FM-15,7,30.02,31,41,0.00,67.0,30.02,29.55,10.00,37.0,230,3.0
3,2024-01-01T02:51:00,FM-15,7,30.03,30,40,0.00,68.0,30.02,29.56,10.00,36.0,240,5.0
4,2024-01-01T03:51:00,FM-15,7,30.01,30,38,0.00,73.0,30.01,29.54,10.00,35.0,240,5.0


# Feature Engineering:

#### Extracting the date & time information:

Each hour is important, but each time component is all in 1 column 'DATE' so it will be more meaningful to put create separate values for each time component: HOUR, DATE, etc.

In [14]:
# convert to datetime
df['DATE'] = pd.to_datetime(df['DATE'])

# extract components
df['year'] = df['DATE'].dt.year
df['month'] = df['DATE'].dt.month
df['day'] = df['DATE'].dt.day

# round to nearest hour
df['DATE_rounded'] = df['DATE'].dt.round('h')

# Extract just the hour (after rounding)
df['hour'] = df['DATE_rounded'].dt.hour

In [15]:
df[['DATE', 'DATE_rounded', 'year', 'month', 'day', 'hour']].tail(25)

,DATE,DATE_rounded,year,month,day,hour
23020,2025-09-28 22:51:00,2025-09-28 23:00:00,2025,9,28,23
23021,2025-09-28 23:51:00,2025-09-29 00:00:00,2025,9,28,0
23022,2025-09-29 00:00:00,2025-09-29 00:00:00,2025,9,29,0
23023,2025-09-29 00:51:00,2025-09-29 01:00:00,2025,9,29,1
23024,2025-09-29 01:00:00,2025-09-29 01:00:00,2025,9,29,1
23025,2025-09-29 01:51:00,2025-09-29 02:00:00,2025,9,29,2
23026,2025-09-29 02:51:00,2025-09-29 03:00:00,2025,9,29,3
23027,2025-09-29 03:51:00,2025-09-29 04:00:00,2025,9,29,4
23028,2025-09-29 04:00:00,2025-09-29 04:00:00,2025,9,29,4
23029,2025-09-29 04:51:00,2025-09-29 05:00:00,2025,9,29,5


In [16]:
df = df.drop(columns=['DATE', 'DATE_rounded'])
df.head()

,REPORT_TYPE,SOURCE,HourlyAltimeterSetting,HourlyDewPointTemperature,HourlyDryBulbTemperature,HourlyPrecipitation,HourlyRelativeHumidity,HourlySeaLevelPressure,HourlyStationPressure,HourlyVisibility,HourlyWetBulbTemperature,HourlyWindDirection,HourlyWindSpeed,year,month,day,hour
0,FM-15,7,30.03,32,42,0.00,68.0,30.03,29.56,10.00,38.0,230,5.0,2024,1,1,1
1,FM-12,4,NaN,32,42,NaN,68.0,30.03,29.57,9.94,38.0,230,5.0,2024,1,1,1
2,FM-15,7,30.02,31,41,0.00,67.0,30.02,29.55,10.00,37.0,230,3.0,2024,1,1,2
3,FM-15,7,30.03,30,40,0.00,68.0,30.02,29.56,10.00,36.0,240,5.0,2024,1,1,3
4,FM-15,7,30.01,30,38,0.00,73.0,30.01,29.54,10.00,35.0,240,5.0,2024,1,1,4


In [17]:
df.columns

Index(['REPORT_TYPE', 'SOURCE', 'HourlyAltimeterSetting',
       'HourlyDewPointTemperature', 'HourlyDryBulbTemperature',
       'HourlyPrecipitation', 'HourlyRelativeHumidity',
       'HourlySeaLevelPressure', 'HourlyStationPressure', 'HourlyVisibility',
       'HourlyWetBulbTemperature', 'HourlyWindDirection', 'HourlyWindSpeed',
       'year', 'month', 'day', 'hour'],
      dtype='object')

In [18]:
df.dtypes

REPORT_TYPE                   object
SOURCE                        object
HourlyAltimeterSetting        object
HourlyDewPointTemperature     object
HourlyDryBulbTemperature      object
HourlyPrecipitation           object
HourlyRelativeHumidity       float64
HourlySeaLevelPressure        object
HourlyStationPressure         object
HourlyVisibility              object
HourlyWetBulbTemperature     float64
HourlyWindDirection           object
HourlyWindSpeed               object
year                           int32
month                          int32
day                            int32
hour                           int32
dtype: object

# Fixing incorrect datatypes of columns

In [19]:
float_cols = [
    'HourlyAltimeterSetting', 'HourlyDewPointTemperature', 'HourlyDryBulbTemperature',
    'HourlyPrecipitation', 'HourlyRelativeHumidity', 'HourlySeaLevelPressure',
    'HourlyStationPressure', 'HourlyVisibility', 'HourlyWetBulbTemperature',
    'HourlyWindDirection', 'HourlyWindSpeed'
]

df[float_cols] = df[float_cols].apply(pd.to_numeric, errors='coerce')

# created a date time object as a column (called ds) for easy inputting into models created by the prophet package
df['ds'] = pd.to_datetime(df[['year', 'month', 'day', 'hour']])

df.dtypes

REPORT_TYPE                          object
SOURCE                               object
HourlyAltimeterSetting              float64
HourlyDewPointTemperature           float64
HourlyDryBulbTemperature            float64
HourlyPrecipitation                 float64
HourlyRelativeHumidity              float64
HourlySeaLevelPressure              float64
HourlyStationPressure               float64
HourlyVisibility                    float64
HourlyWetBulbTemperature            float64
HourlyWindDirection                 float64
HourlyWindSpeed                     float64
year                                  int32
month                                 int32
day                                   int32
hour                                  int32
ds                           datetime64[ns]
dtype: object

#### Filling in missing values:

In [20]:
# missing values in each column
print("missing values per column:")
df.isna().sum()


missing values per column:


REPORT_TYPE                     0
SOURCE                        280
HourlyAltimeterSetting       5505
HourlyDewPointTemperature     642
HourlyDryBulbTemperature      647
HourlyPrecipitation          9948
HourlyRelativeHumidity        638
HourlySeaLevelPressure       3194
HourlyStationPressure        2467
HourlyVisibility              709
HourlyWetBulbTemperature     2464
HourlyWindDirection          2762
HourlyWindSpeed               649
year                            0
month                           0
day                             0
hour                            0
ds                              0
dtype: int64

In [21]:
# fill missing values differently based on data types of columns
for col in df.columns:
    if df[col].isna().sum() > 0:  # only process columns with NaNs
        
        if np.issubdtype(df[col].dtype, np.floating):
            df[col] = df[col].fillna(df[col].mean()) # mean for float values
            
        elif np.issubdtype(df[col].dtype, np.integer):
            df[col] = df[col].fillna(df[col].median()) # median for int values
            
        else:  # object, category, string, etc.
            mode_val = df[col].mode() # mode for categorical values
            if not mode_val.empty:
                df[col] = df[col].fillna(mode_val[0])


In [22]:
# verify results
print("\nRemaining missing values:")
print(df.isna().sum().sum())  # total number of NaNs left


Remaining missing values:
0


In [23]:
df.head()

,REPORT_TYPE,SOURCE,HourlyAltimeterSetting,HourlyDewPointTemperature,HourlyDryBulbTemperature,HourlyPrecipitation,HourlyRelativeHumidity,HourlySeaLevelPressure,HourlyStationPressure,HourlyVisibility,HourlyWetBulbTemperature,HourlyWindDirection,HourlyWindSpeed,year,month,day,hour,ds
0,FM-15,7,30.030000,32.0,42.0,0.000000,68.0,30.03,29.56,10.00,38.0,230.0,5.0,2024,1,1,1,2024-01-01 01:00:00
1,FM-12,4,449.761837,32.0,42.0,0.138193,68.0,30.03,29.57,9.94,38.0,230.0,5.0,2024,1,1,1,2024-01-01 01:00:00
2,FM-15,7,30.020000,31.0,41.0,0.000000,67.0,30.02,29.55,10.00,37.0,230.0,3.0,2024,1,1,2,2024-01-01 02:00:00
3,FM-15,7,30.030000,30.0,40.0,0.000000,68.0,30.02,29.56,10.00,36.0,240.0,5.0,2024,1,1,3,2024-01-01 03:00:00
4,FM-15,7,30.010000,30.0,38.0,0.000000,73.0,30.01,29.54,10.00,35.0,240.0,5.0,2024,1,1,4,2024-01-01 04:00:00


# Handling duplicate date values:

For Prophet (and time-series modeling in general), each timestamp (ds) should be unique. Duplicate timestamps will confuse Prophet’s fitting process, cause errors like “non-unique ds values”, or distort trend and seasonality estimates.

It made more sense to take the average of duplicate values for each hour.

In [24]:
# check if there are no duplicate timestamps in 'ds'

dup_vals = df[df['ds'].duplicated(keep=False)]
print(f"Number of duplicate timestamps: {dup_vals['ds'].nunique()}")

Number of duplicate timestamps: 6560


In [25]:
dup_vals.head()

,REPORT_TYPE,SOURCE,HourlyAltimeterSetting,HourlyDewPointTemperature,HourlyDryBulbTemperature,HourlyPrecipitation,HourlyRelativeHumidity,HourlySeaLevelPressure,HourlyStationPressure,HourlyVisibility,HourlyWetBulbTemperature,HourlyWindDirection,HourlyWindSpeed,year,month,day,hour,ds
0,FM-15,7,30.030000,32.0,42.0,0.000000,68.0,30.03,29.56,10.00,38.0,230.0,5.0,2024,1,1,1,2024-01-01 01:00:00
1,FM-12,4,449.761837,32.0,42.0,0.138193,68.0,30.03,29.57,9.94,38.0,230.0,5.0,2024,1,1,1,2024-01-01 01:00:00
4,FM-15,7,30.010000,30.0,38.0,0.000000,73.0,30.01,29.54,10.00,35.0,240.0,5.0,2024,1,1,4,2024-01-01 04:00:00
5,FM-12,4,449.761837,30.0,38.0,0.138193,73.0,30.01,29.55,9.94,35.0,240.0,5.0,2024,1,1,4,2024-01-01 04:00:00
8,FM-15,7,30.030000,32.0,40.0,0.000000,73.0,30.03,29.56,10.00,37.0,230.0,5.0,2024,1,1,7,2024-01-01 07:00:00


```
df = df.groupby('ds', as_index=False).mean(numeric_only=True)
```

Groups the Data Frame by the ds column (datetime). For duplicate timestamps, it finds the mean of all numeric columns ignoring non-numeric ones. It keeps only one row per unique ds.

In [26]:
non_num = df[['ds', 'SOURCE', 'REPORT_TYPE']].drop_duplicates('ds')
num_avg = df.groupby('ds', as_index=False).mean(numeric_only=True)
df = num_avg.merge(non_num, on='ds', how='left')


In [27]:
# verify there are no more duplicate timestamps in 'ds'

dup_vals = df[df['ds'].duplicated(keep=False)]
print(f"Number of duplicate timestamps: {dup_vals['ds'].nunique()}")

Number of duplicate timestamps: 0


In [28]:
df.head(30)

,ds,HourlyAltimeterSetting,HourlyDewPointTemperature,HourlyDryBulbTemperature,HourlyPrecipitation,HourlyRelativeHumidity,HourlySeaLevelPressure,HourlyStationPressure,HourlyVisibility,HourlyWetBulbTemperature,HourlyWindDirection,HourlyWindSpeed,year,month,day,hour,SOURCE,REPORT_TYPE
0,2024-01-01 00:00:00,239.955918,28.898257,41.982568,0.069097,59.83137,238.854888,231.777947,10.794917,36.739289,68.318050,5.352951,2024.0,1.0,1.0,0.0,7,FM-15
1,2024-01-01 01:00:00,239.895918,32.000000,42.000000,0.069097,68.00000,30.030000,29.565000,9.970000,38.000000,230.000000,5.000000,2024.0,1.0,1.0,1.0,7,FM-15
2,2024-01-01 02:00:00,30.020000,31.000000,41.000000,0.000000,67.00000,30.020000,29.550000,10.000000,37.000000,230.000000,3.000000,2024.0,1.0,1.0,2.0,7,FM-15
3,2024-01-01 03:00:00,30.030000,30.000000,40.000000,0.000000,68.00000,30.020000,29.560000,10.000000,36.000000,240.000000,5.000000,2024.0,1.0,1.0,3.0,7,FM-15
4,2024-01-01 04:00:00,239.885918,30.000000,38.000000,0.069097,73.00000,30.010000,29.545000,9.970000,35.000000,240.000000,5.000000,2024.0,1.0,1.0,4.0,7,FM-15
5,2024-01-01 05:00:00,30.010000,30.000000,39.000000,0.000000,70.00000,30.010000,29.540000,10.000000,35.000000,250.000000,3.000000,2024.0,1.0,1.0,5.0,7,FM-15
6,2024-01-01 06:00:00,30.020000,31.000000,39.000000,0.000000,73.00000,30.020000,29.550000,10.000000,36.000000,230.000000,7.000000,2024.0,1.0,1.0,6.0,7,FM-15
7,2024-01-01 07:00:00,239.895918,32.000000,40.000000,0.069097,73.00000,30.030000,29.565000,9.970000,37.000000,230.000000,5.000000,2024.0,1.0,1.0,7.0,7,FM-15
8,2024-01-01 08:00:00,30.030000,32.000000,41.000000,0.000000,70.00000,30.040000,29.560000,10.000000,37.000000,240.000000,3.000000,2024.0,1.0,1.0,8.0,7,FM-15
9,2024-01-01 09:00:00,30.050000,33.000000,43.000000,0.000000,68.00000,30.050000,29.580000,10.000000,39.000000,230.000000,6.000000,2024.0,1.0,1.0,9.0,7,FM-15


# Exporting the final data to a CSV file:

In [29]:
df.to_csv('clean_wthr_dt.csv', index = False)